fs/mmap/fs_munmap.c

```diff
@@ -158,7 +159,40 @@ static int file_munmap_(FAR void *start, size_t length, bool kernel)
   nxmutex_unlock(&g_rammaps.lock);
   return ret;
 #else
-  return OK;
+
+  FAR struct tcb_s *tcb = nxsched_self();
+  FAR struct task_group_s *group = tcb->group;
+  FAR struct mm_map_entry_s *entry = NULL;
+  int ret = OK;
+
+  /* Iterate through all the mappings and call the underlying
+   * unmap for every mapping where "start" lies
+   * break loop on any errors.
+   *
+   * Get exclusive access to mm_map for this
+   */
+
+  ret = mm_map_lock();
+  if (ret == OK)
+    {
+      while (ret == OK && (entry = mm_map_find(start, length)))
+        {
+          if (entry->munmap)
```

Comment:
let's debug assert, it's programming error not runtime error if entry->munmap is NULL

Reply:
sure, done